src/hipscat/inspection/almanac_catalog_info.py

```diff
@@ -0,0 +1,149 @@
+                                                          from_catalog_dir)
+from hipscat.io import file_io
+
+
+@dataclass
+class AlmanacCatalogInfo:
+    """Container for parsed almanac information."""
+
+    file_path: str = ""
+    namespace: str = ""
+    catalog_path: str = ""
+    catalog_name: str = ""
+    catalog_type: str = ""
+    primary: str = ""
+    join: str = ""
+    primary_link: Self = None
+    join_link: Self = None
+    sources: List[Self] = field(default_factory=list)
+    objects: List[Self] = field(default_factory=list)
+    margins: List[Self] = field(default_factory=list)
+    associations: List[Self] = field(default_factory=list)
+    associations_right: List[Self] = field(default_factory=list)
+    indexes: List[Self] = field(default_factory=list)
+
+    creators: List[str] = field(default_factory=list)
+    description: str = ""
+    version: str = ""
+    deprecated: str = ""
+
+    catalog_info: dict = field(default_factory=dict)
+
+    catalog_info_object: BaseCatalogInfo = None
+
+    def __post_init__(
+        self,
+    ):
+        if len(self.catalog_info):
+            self.catalog_info_object = create_catalog_info(self.catalog_info)
+
+        ## Allows use of $HIPSCAT_DEFAULT_DIR in paths
+        self.catalog_path = os.path.expandvars(self.catalog_path)
+
+    def get_primary_text(self) -> str:
+        """Find some text name/path for a primary catalog link, if present"""
+        if self.primary:
+            return self.primary
+        if self.catalog_info and "primary_catalog" in self.catalog_info:
+            return self.catalog_info["primary_catalog"]
+        return None
+
+    def get_join_text(self) -> str:
+        """Find some text name/path for a join catalog link, if present"""
+        if self.join:
+            return self.join
+        if self.catalog_info and "join_catalog" in self.catalog_info:
+            return self.catalog_info["join_catalog"]
+        return None
+
+    @staticmethod
+    def get_default_dir() -> str:
+        """Fetch the default directory for environment variables.
+
+        This is set via the environment variable: HIPSCAT_ALMANAC_DIR
+
+        To set this in a linux-like environment, use a command like:
+
+            export HIPSCAT_ALMANAC_DIR=/data/path/to/almanacs
+
+        This will also attempt to expand any environment variables WITHIN the default
+        directory environment variable. This can be useful in cases where:
+
+            $HIPSCAT_ALMANAC_DIR=$HIPSCAT_DEFAULT_DIR/almanacs/
+        """
+        default_dir = os.getenv("HIPSCAT_ALMANAC_DIR", "")
+        if default_dir:
+            default_dir = os.path.expandvars(default_dir)
+        return default_dir
+
+    @classmethod
+    def from_catalog_dir(cls, catalog_base_dir: str) -> Self:
+        """Create almanac information from the catalog information found at the target directory"""
+        catalog_info = from_catalog_dir(catalog_base_dir=catalog_base_dir)
```

Comment:
Could this method be imported with some kind of prefix like `catalog_info_factory.from_catalog_dir`? It's a bit confusing that its the same name as the method being defined.